hw/ip/rom_ctrl/dv/cov/rom_ctrl_cov_excl.el

```diff
@@ -38,3 +38,87 @@ ANNOTATION: "Waived the assignment as the input is wired to a parameter"
+// inside the instantiation of tlul_adapter_sram in rom_ctrl.sv. Hence, if wr_attr_error
+// is true then tl_i.a_opcode is PutFullData or PutPartialData which automatically sets
+// wr_vld_error.
+Condition 17 "3469950311" "(wr_attr_error | wr_vld_error | rd_vld_error | instr_error | tlul_error |
+                            intg_error) 1 -1" (7 "100000")
+
+// Because en_ifetch_i is hardwired to MuBi4True, instr_error can only be true if
+// tl_i.a_user.instr_type is not a valid mubi value. But that gets seen in u_err as instr_type_err,
+// causing tlul_error to be true in the adapter.
+Condition 17 "3469950311" "(wr_attr_error | wr_vld_error | rd_vld_error | instr_error | tlul_error |
+                            intg_error) 1 -1" (4 "000100")
+
+// For rspfifo_rdata.error, it is pushed in rspfifo_wdata.error. If the error field is true, it
+// means rerror_i[1] bit is true but the rerror_i bits are wired to 0 in the instantiation of the
+// tlul_adapter_sram in rom_ctrl.sv.
+Condition 4 "3638058042" "(rspfifo_rdata.error | reqfifo_rdata.error) 1 -1" (3 "10")
+
+// The 011 case is not possible. If d_valid is false then we must either have reqfifo_rvalid = 0 or
+// rspfifo_rvalid = 0 (with a response to Read). The second two terms imply that rspfifo_rvalid = 1,
+// so we must have that reqfifo_rvalid is false. That means that u_reqfifo is empty.
+//
+// For rspfifo_rvalid to be true, we must either have a response arriving now (since Pass=1) or have
+// a stored response. In the first case, we must also have reqfifo_rvalid = 1, which is checked with
+// the rvalidHighReqFifoEmpty assertion. For the second case, we must have had the request in the
+// fifo and popped it before now. But this isn't possible, because the ROM always responds in a
+// single cycle: the same cycle we would be popping the request from u_reqfifo.
+Condition 21 "3623514242" "(d_valid & rspfifo_rvalid & (reqfifo_rdata.op == OpRead)) 1 -1" (1 "011")
+
+// Condition 24 till 34 are excluded for the same reason. The requirements for the conditions are:
+// 1) reqfifo_rvalid --> 1
+// 2) rspfifo_rvalid --> 1
+// 3) reqfifo_rdata.op --> opread
+// 4) reqfifo_rdata.error --> 1
+//
+// We cannot see vld_rd_rsp && d_error. If d_error is true, we must have rspfifo_rdata.error or
+// reqfifo_rdata.error. The first cannot happen, because the items in the response fifo have an
+// error signal from rerror_i[1] and rom_ctrl connects rerror_i to zero. For the second to happen,
+// we must have pushed in an item with error = 1, which means that error_internal was set. But that
+// will have squashed the request, so req_o was low. As a result, the request fifo item will be
+// popped again before anything appears in the response fifo.
+Condition 24 "1059982851" "(vld_rd_rsp & ((~d_error))) 1 -1" (2 "10")
+
+// The reason for condition 24 explains why we can't get (vld_rd_rsp && reqfifo_rdata.error) ---> 1
```

Comment:
This text looks good to me, but should we use "=" instead of "--->"? (I'm not sure I know what "--->" means!)